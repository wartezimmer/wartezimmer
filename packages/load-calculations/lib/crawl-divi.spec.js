const { readFileSync } = require('fs');
const path = require('path');
const knex = require('knex');
const { logger } = require('./logger')

const { extractDataPerClinicFromHtml, crawlIntensiveCareRegister } = require('./crawl-divi');

describe('Crawl Divi', () => {
    jest.setTimeout(15000)

    let db = null

    beforeAll(async () => {
        db = knex({
            client: 'pg',
            connection: process.env.TEST_DATABASE_URL,
            log: logger,
            acquireConnectionTimeout: 10000,
        })
        
        await db('divi_icu_register').truncate()
    });

    afterAll(async () => {
        db.destroy()
    });

    it('extracts data from html', () => {
        const html = readFileSync(path.resolve(__dirname, '../test/data/https___www.divi.de_register_intensivregister_list[limit]=10.html'))
        const data = extractDataPerClinicFromHtml(html, 'Mon, 30 Mar 2020 09:38:59 GMT');

        expect(data).toMatchSnapshot();
    });

    // TODO: use jest-nock package to mock the request
    // -> currently skipped until mocked 
    it.skip('crawls data from an intensive care url and writes it to db', async () => {
        await crawlIntensiveCareRegister(db, 'https://www.divi.de/register/intensivregister?list[limit]=0', 'Mon, 30 Mar 2020 09:38:59 GMT');

        const dbData = await db('divi_icu_register').select('*')
        const dbDataWithoutCreatedAt = dbData.map((item) => ({ ...item, createdAt: null }))
        
        expect(dbDataWithoutCreatedAt).toMatchSnapshot()
    });
});
