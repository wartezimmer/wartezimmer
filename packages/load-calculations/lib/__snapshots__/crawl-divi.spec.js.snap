// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl Divi crawls data from an intensive care url and writes it to db 1`] = `
Array [
  Object {
    "address_city": "Berlin",
    "address_housenumber": "63",
    "address_postcode": "14109",
    "address_state": "BE",
    "address_station": "Anästhesie",
    "address_street": "Königstraße",
    "contact_station": null,
    "contact_website": "Http://www.berlin.immanuel.de",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 1,
    "last_update_raw": "30.03.2020 13:36",
    "name": "Immanuel-Krankenhaus GmbH",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Burg",
    "address_housenumber": "55a",
    "address_postcode": "39288",
    "address_state": "ST",
    "address_station": "Anästhesie und Intensivmedizin",
    "address_street": "August-Bebel-Straße",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.helios-gesundheit.de/kliniken/burg",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 2,
    "last_update_raw": "30.03.2020 13:22",
    "name": "Helios Klinik Jerichower Land GmbH",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Hamburg",
    "address_housenumber": "52",
    "address_postcode": "20251",
    "address_state": "HH",
    "address_station": "Sektion Neonatologie und Pädiatrische Intensivmedizin",
    "address_street": "Martinistraße",
    "contact_station": "Kinderintensivstation K1a (Kinder-UKE)",
    "contact_website": "https://www.uke.de/kliniken-institute/kliniken/kinder-und-jugendmedizin/neonatologie-und-p%C3%A4diatrische-intensivmedizin/index.html",
    "createdAt": null,
    "ecmo_status": 2,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 3,
    "last_update_raw": "30.03.2020 13:26",
    "name": "Universitätsklinikum Hamburg-Eppendorf",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Uelzen",
    "address_housenumber": "34",
    "address_postcode": "29525",
    "address_state": "NI",
    "address_station": "Klinik für Anästhesiologie, Intensivmedizin, Notfallmedizin und Schmerztherapie",
    "address_street": "Hagenskamp",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.helios-gesundheit.de/kliniken/uelzen/",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 4,
    "last_update_raw": "30.03.2020 13:12",
    "name": "Helios- Klinikum Uelzen",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Borstel",
    "address_housenumber": "35",
    "address_postcode": "23845",
    "address_state": "SH",
    "address_station": "Innere Medizin",
    "address_street": "Parkallee",
    "contact_station": "Intensivstation",
    "contact_website": "https://klinik.fz-borstel.de",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 5,
    "last_update_raw": "30.03.2020 13:09",
    "name": "Medizinische Klinik Borstel, Forschungszentrum Borstel",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "München",
    "address_housenumber": "71a",
    "address_postcode": "80639",
    "address_state": "BY",
    "address_station": "1. Medizinische Abteilung",
    "address_street": "Renatastrasse",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.krankenhaus-neuwittelsbach.de/",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 3,
    "icu_low_status": 1,
    "id": 6,
    "last_update_raw": "30.03.2020 13:00",
    "name": "Krankenhaus Neuwittelsbach",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Nördlingen",
    "address_housenumber": "4",
    "address_postcode": "86720",
    "address_state": "BY",
    "address_station": "Kardiologie",
    "address_street": "Stoffelberg",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.donkliniken.de/index.php/donkliniken/Patienten-und-Besucher/Unsere-Kliniken/Standort-Noerdlingen",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 7,
    "last_update_raw": "30.03.2020 13:05",
    "name": "Stiftungskrankenhaus Nördlingen",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Schwerin",
    "address_housenumber": "397",
    "address_postcode": "19049",
    "address_state": "MV",
    "address_station": "Klinik f. Intensivmedizin",
    "address_street": "Wismarsche Straße",
    "contact_station": "B2   I2  E3",
    "contact_website": "https://www.helios-gesundheit.de/schwerin/",
    "createdAt": null,
    "ecmo_status": 1,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 8,
    "last_update_raw": "30.03.2020 13:32",
    "name": "Helios Kliniken Schwerin",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Damp",
    "address_housenumber": "20",
    "address_postcode": "24351",
    "address_state": "SH",
    "address_station": "Anästhesie und Intensivmedizin",
    "address_street": "Seute Deern Ring",
    "contact_station": "Intensiv 4B",
    "contact_website": "http://vamed-gesundheit.de",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 9,
    "last_update_raw": "30.03.2020 12:51",
    "name": "Vamed Ostseeklinik Damp",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
  Object {
    "address_city": "Haßfurt",
    "address_housenumber": "69",
    "address_postcode": "97437",
    "address_state": "BY",
    "address_station": null,
    "address_street": "Hofheimerstr.",
    "contact_station": "Intensivstation",
    "contact_website": "http://www.hassberg-kliniken.de",
    "createdAt": null,
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "id": 10,
    "last_update_raw": "30.03.2020 12:50",
    "name": "Haßberg-Kliniken Haus Haßfurt",
    "run_time": 2020-03-30T07:38:59.000Z,
  },
]
`;

exports[`Crawl Divi extracts data from html 1`] = `
Array [
  Object {
    "address_city": "Münster",
    "address_housenumber": "124",
    "address_postcode": "48143",
    "address_state": "NRW",
    "address_station": null,
    "address_street": "Düesbergweg",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.clemenshospital.de/",
    "ecmo_status": 2,
    "icu_high_status": 2,
    "icu_low_status": 2,
    "last_update_raw": "29.03.2020 13:27",
    "name": "Clemenshospital Münster",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Köln",
    "address_housenumber": "445-449",
    "address_postcode": "50933",
    "address_state": "NRW",
    "address_station": "Anästhesiologie und Intensivmedizin",
    "address_street": "Aachener Straße",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.sana.de/koeln",
    "ecmo_status": 0,
    "icu_high_status": 3,
    "icu_low_status": 3,
    "last_update_raw": "29.03.2020 11:39",
    "name": "Sana Dreifaltigkeits-Krankenhaus Köln",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "(Spreewald)",
    "address_housenumber": "29",
    "address_postcode": "15907 Lübben",
    "address_state": "BB",
    "address_station": "Anä / IT",
    "address_street": "Schillerstraße",
    "contact_station": "ITS",
    "contact_website": "http://https//klinikum-ds.de",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "29.03.2020 10:40",
    "name": "Spreewaldklinik Lübben",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Finsterwalde",
    "address_housenumber": "42",
    "address_postcode": "3238",
    "address_state": "BB",
    "address_station": "Anästhesie",
    "address_street": "Kichhainer Str.",
    "contact_station": "Intensivtation Standort Finsterwalde",
    "contact_website": "http://www.elbe-elster-klinikum.de",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "28.03.2020 15:24",
    "name": "Elbe-Elster-Klinikum, Standort Finsterwalde",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Wolfratshausen",
    "address_housenumber": "5",
    "address_postcode": "82515",
    "address_state": "BY",
    "address_station": null,
    "address_street": "Moosbauerweg",
    "contact_station": null,
    "contact_website": "http://www.kreisklinik-wolfratshausen.de",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "29.03.2020 08:32",
    "name": "Kreisklinik Wolfratshausen",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Friedrichshafen",
    "address_housenumber": "2",
    "address_postcode": "88048",
    "address_state": "BW",
    "address_station": null,
    "address_street": "Röntgenstraße",
    "contact_station": null,
    "contact_website": "http://medizin-campus-bodensee.de",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "29.03.2020 13:54",
    "name": "Medizin Campus Bodensee - Klinikum Friedrichshafen",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Bremen",
    "address_housenumber": "228",
    "address_postcode": "28755",
    "address_state": "HB",
    "address_station": "Klinik für Anästhesiologie, Intensiv- und Notfallmedizin",
    "address_street": "Hammersbecker Straße",
    "contact_station": "Intensivstation",
    "contact_website": "https://www.gesundheitnord.de/krankenhaeuserundzentren/kbn/klinikum-bremen-nord/anaesthesieundintensivmedizinkbn.html",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "29.03.2020 12:11",
    "name": "Klinikum-Bremen-Nord",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Berlin",
    "address_housenumber": "3-5",
    "address_postcode": "12207",
    "address_state": "BE",
    "address_station": "Anästhesie und Intensivmedizin",
    "address_street": "Promenadenstraße",
    "contact_station": "ITS",
    "contact_website": "https://www.bethelnet.de",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "29.03.2020 14:29",
    "name": "Krankenhaus Bethel Berlin",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Kronach",
    "address_housenumber": "5a",
    "address_postcode": "96317",
    "address_state": "BY",
    "address_station": "Anästhesie",
    "address_street": "Kestel",
    "contact_station": "4a",
    "contact_website": "https://www.helios-gesundheit.de/kliniken/kronach-frankenwaldklinik/",
    "ecmo_status": 0,
    "icu_high_status": 2,
    "icu_low_status": 2,
    "last_update_raw": "27.03.2020 17:00",
    "name": "Helios Frankenwaldklinik Kronach",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Coesfeld",
    "address_housenumber": "41",
    "address_postcode": "48653",
    "address_state": "NRW",
    "address_station": null,
    "address_street": "Südring",
    "contact_station": "cD1  Interdisziplinäre Intensivstation",
    "contact_website": "https://www.christophorus-kliniken.de/",
    "ecmo_status": 0,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "27.03.2020 16:42",
    "name": "Christophoruskliniken Coesfeld",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Vechta",
    "address_housenumber": "6-8",
    "address_postcode": "49377",
    "address_state": "NI",
    "address_station": "Abteilung für Anästhesie und Intensivmedizin",
    "address_street": "Marienstraße",
    "contact_station": "Interdisziplinäre Intensivstation",
    "contact_website": "https://www.marienhospital-vechta.de/leistungen/anaesthesie-intensivmedizin.html",
    "ecmo_status": 3,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "17.03.2020 18:41",
    "name": "St. Marienhospital Vechta gGmbH",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
  Object {
    "address_city": "Vechta",
    "address_housenumber": "6-8",
    "address_postcode": "49377",
    "address_state": "NI",
    "address_station": "Abteilung für Anästhesie und Intensivmedizin",
    "address_street": "Marienstraße",
    "contact_station": "Interdisziplinäre Intensivstation",
    "contact_website": "https://www.marienhospital-vechta.de/leistungen/anaesthesie-intensivmedizin.html",
    "ecmo_status": 3,
    "icu_high_status": 1,
    "icu_low_status": 1,
    "last_update_raw": "17.03.2020 18:41",
    "name": "St. Marienhospital Vechta gGmbH (2)",
    "run_time": "Mon, 30 Mar 2020 09:38:59 GMT",
  },
]
`;
